From ef4f2357df6e9dd6213ce00d01253f3c458d5df0 Mon Sep 17 00:00:00 2001
From: Randy Dunlap <rdunlap@infradead.org>
Date: Thu, 23 Jul 2026 11:12:15 -0700
Subject: [PATCH 06827/15333] regmap: clean up kernel-doc comments

- add/correct missing struct members in struct regmap_bus
- add a missing function parameter @dev
- add missing struct member in struct regmap_irq_type
- add missing struct members in struct regmap_irq_chip

to prevent kernel-doc warnings:

Warning: include/linux/regmap.h:630 struct member 'reg_noinc_write' not
 described in 'regmap_bus'
Warning: include/linux/regmap.h:630 struct member 'reg_noinc_read' not
 described in 'regmap_bus'
Warning: include/linux/regmap.h:630 Excess struct member 'reg_write_noinc'
 description in 'regmap_bus'
Warning: include/linux/regmap.h:1001 function parameter 'dev' not
 described in 'regmap_init_sdw_mbq_cfg'
Warning: include/linux/regmap.h:1600 struct member 'type_reg_mask' not
 described in 'regmap_irq_type'
Warning: include/linux/regmap.h:1775 struct member 'irq_reqres' not
 described in 'regmap_irq_chip'
Warning: include/linux/regmap.h:1775 struct member 'irq_relres' not
 described in 'regmap_irq_chip'

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Mark Brown <broonie@kernel.org>
---
 include/linux/regmap.h | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/include/linux/regmap.h b/include/linux/regmap.h
index 370baa19db87..c8aebd148e08 100644
--- a/include/linux/regmap.h
+++ b/include/linux/regmap.h
@@ -587,7 +587,7 @@ typedef void (*regmap_hw_free_context)(void *context);
  *               must serialise with respect to non-async I/O.
  * @reg_write: Write a single register value to the given register address. This
  *             write operation has to complete when returning from the function.
- * @reg_write_noinc: Write multiple register value to the same register. This
+ * @reg_noinc_write: Write multiple register values to the same register. This
  *             write operation has to complete when returning from the function.
  * @reg_update_bits: Update bits operation to be used against volatile
  *                   registers, intended for devices supporting some mechanism
@@ -596,6 +596,8 @@ typedef void (*regmap_hw_free_context)(void *context);
  * @read: Read operation.  Data is returned in the buffer used to transmit
  *         data.
  * @reg_read: Read a single register value from a given register address.
+ * @reg_noinc_read: Read multiple register values from the same register. This
+ *             read operation has to complete when returning from the function.
  * @free_context: Free context.
  * @async_alloc: Allocate a regmap_async() structure.
  * @read_flag_mask: Mask to be set in the top byte of the register when doing
@@ -991,7 +993,8 @@ bool regmap_ac97_default_volatile(struct device *dev, unsigned int reg);
 /**
  * regmap_init_sdw_mbq_cfg() - Initialise MBQ SDW register map with config
  *
- * @sdw: Device that will be interacted with
+ * @dev: &struct device that will be interacted with
+ * @sdw: Soundwire device that will be interacted with
  * @config: Configuration for register map
  * @mbq_config: Properties for the MBQ registers
  *
@@ -1584,6 +1587,7 @@ regmap_fields_force_update_bits(struct regmap_field *field, unsigned int id,
  * struct regmap_irq_type - IRQ type definitions.
  *
  * @type_reg_offset: Offset register for the irq type setting.
+ * @type_reg_mask: Device interrupt mask
  * @type_rising_val: Register value to configure RISING type irq.
  * @type_falling_val: Register value to configure FALLING type irq.
  * @type_level_low_val: Register value to configure LEVEL_LOW type irq.
@@ -1717,6 +1721,8 @@ struct regmap_irq_chip_data;
  *		 main status base, [0, num_config_regs[ for any config
  *		 register base, and [0, num_regs[ for any other base.
  *		 If unspecified then regmap_irq_get_irq_reg_linear() is used.
+ * @irq_reqres:	 Callback function to request IRQ resources (may be %NULL)
+ * @irq_relres:  Callback function to release IRQ resources (may be %NULL)
  * @irq_drv_data:    Driver specific IRQ data which is passed as parameter when
  *		     driver specific pre/post interrupt handler is called.
  *
-- 
2.55.0

